app/Http/Controllers/Legacy/Web/ActionsController.php

```diff
@@ -28,13 +46,16 @@ public function store(Request $request)
             'name' => 'required|string',
             'campaign_id' => 'required|integer|exists:campaigns,id',
             'post_type' => 'required|string',
-            'reportback' => 'required|boolean',
-            'civic_action' => 'required|boolean',
-            'scholarship_entry' => 'required|boolean',
             'noun' => 'required|string',
             'verb' => 'required|string',
         ]);
 
+        // Checkbox values are only sent from the front end if they are checked.
+        // Assign checkbox values if sent from the front end or are 1 (sent from API).
+        $request['reportback'] = isset($request['reportback']) || $request['reportback'] === 1 ? 1 : 0;
```

Comment:
Could the api send `true`? I don't think that is `===` to ` 1`

Reply:
@katiecrane it expects a boolean! Should it be `==` instead of `===`?

Reply:
I believe `===` will hold it to the exact type and value of `1`. Since it's going to be a boolean could you use `$request['reportback']` itself? If the value as `false` it'd get evaluated to `false` right?

Reply:
@katiecrane you're totally right, changed to `==` instead! 

The reason why we can't just use `$request['reportback']` itself is that it will only get send in the payload from the form if it is checked. If it is not checked, `$request['reportback']` won't be in the payload at all, so that is why we have to do this weird logic in order for it to register unchecked values as well.

Reply:
ah sorry @chloealee I was unclear! I mean something like this:
```
$request['scholarship_entry'] = isset($request['scholarship_entry']) || $request['scholarship_entry'] ? 1 : 0;
```

Reply:
@katiecrane ah thank you! I just tried that via the api and it didn't work :-/ `$request['scholarship_entry']` came in as `"0"` via Postman so since that was a truthy value, it saved `1` in the database incorrectly for it to count as a scholarship.

Reply:
hmm that's so strange since we make sure that it is a boolean. since that is a string would it still come through as `true` in your updated way with `== 1`?

Reply:
it's weird that it is coming in as a string but in the request in Postman, I am putting `0` as the value, not `false`. So with the updated `==`, since it doesn't look for type, it works!

Reply:
oohhh I see what you're saying now! Ok got it working! I changed `||` to `&&` in this ternary and I think this makes more sense!

Reply:
YES I LOVE THAT